### ISSUE TITLE
Having multiple Kinto instances in the same origin

### ISSUE THREAD

**Issue by michielbdejong** [Kinto/kinto.js#119]
It [seem that](https://github.com/Kinto/kinto.js/blob/master/src/collection.js#L63) the names of IndexedDB databases used only depends on the bucket name and the collection name. That means that if I want to create two Kinto instances, if I query a collection on instance 1, it will also return all the items that are in the collection with the same name in instance 2?

Is this by design? If so, then maybe it would be clearer to define one global Kinto instance, so that instead of:

````js
var myInstance = new Kinto(options);
myInstance.collection('history').list().etc....
````
we could just do:

````js
Kinto.configure(options);
Kinto.collection('history').list().etc....
````
Or another option (if multiple instance in the same origin are desirable), add a `dbNamePrefix` option, so then you could do:
````js
var instance1 = new Kinto({ dbNamePrefix: 'instance1' });
var instance2 = new Kinto({ dbNamePrefix: 'instance2' });
````
and it would be safe.

**Comment by michielbdejong** [comment]
Oh never mind, the `bucket` option does exactly that, of course. :)

**Comment by michielbdejong** [comment]
Ah no, it doesn't, because the bucket is also part of the URL it will retrieve.

**Comment by n1k0** [comment]
I don't clearly understand what the problem actually is with the current behavior; a Kinto instance allow accessing a given database identified by a bucket name, them collections by their names; it will always fetch and persist data in a unique matching local database. 

I feel I'm probably missing something obvious here, hint?

**Comment by michielbdejong** [comment]
Hm, let me try to invent an example: Let's say I have two Sync accounts, and want to use Syncto on localhost to compare which bookmarks I have in each of them. If I do:

````js
var kinto1 = new Kinto({
  bucket: 'syncto',
  remote: 'http://localhost:8000/v1/',
  headers: {
    'Authorization': 'BrowserID ' + assertion1,
    'X-Client-State': xClientState1
  }
});
var kinto2 = new Kinto({
  bucket: 'syncto',
  remote: 'http://localhost:8000/v1/',
  headers: {
    'Authorization': 'BrowserID ' + assertion2,
    'X-Client-State': xClientState2
  }
});
kinto1.collection('bookmarks').sync().then(function {
  return kinto1.collection('bookmarks').list();
}).then(function(list) {
  console.log('Account 1 contains these:', list);
}).then(function() {
  return kinto2.collection('bookmarks').sync();
}).then(function {
  return kinto2.collection('bookmarks').list();
}).then(function(list) {
  console.log('Account 2 contains these:', list);
});
````

Would this code work? Or would both `kinto1` and `kinto2` store the data in a database called `'syncto/bookmarks'` and thus clash?

**Comment by n1k0** [comment]
Oh I see. In your example, both instances will indeed store distinct users data in the very same database, which is unwanted. @leplatrem has been working on #111 to add support for prefixing a db name, maybe that could solve your issue?

Did you have any other approach in mind?

**Comment by Natim** [comment]
A proposed approched was to hash the kinto server and use it as a prefix, but I think it is what @leplatrem planned first and then he said it is better to be explicit on the prefix.

**Comment by michielbdejong** [comment]
@Natim note that in my example, the kinto URL is always 'http://localhost:8000/v1/', so that wouldn't be enough there.

**Comment by Natim** [comment]
Sorry the proposal was baseURL + Authorization header.

**Comment by Natim** [comment]
In your case if you use the client state as a prefix it would work nicely.

**Comment by n1k0** [comment]
Can we declare this fixed now #111 has landed? @michielbdejong @Natim

**Comment by michielbdejong** [comment]
Yes, we can! Thanks a lot for that! I'll use the logged in user's email address as a dbPrefix.
